qualang_tools/loops/loops.py

```diff
@@ -7,23 +7,25 @@
     - qua_logspace: Function parametrizing the QUA `for_` loop from the numpy.logspace() syntax.
 """
 import numpy as np
+import warnings
+from qm.qua import Cast
 
 
-def from_array(var, array, var_type):
+def from_array(var, array):
     """Function parametrizing the QUA `for_` loop from a python array.
 
     :param var: the QUA variable that will be looped over (int or fixed).
-    :param array: a Python list or numpy array containing the values over which `a` will be looping. The spacing must be constant or logarithmic and it must be a Python array.
-    :param var_type: type of the QUA variable 'a', can be either 'int' or 'fixed'.
+    :param array: a Python list or numpy array containing the values over which `a` will be looping. The spacing must be even in linear or logarithmic scales and it cannot be a QUA array.
     :return: QUA for_ loop parameters (var, init, cond, update) as defined in https://qm-docs.qualang.io/api_references/qua/dsl_main?highlight=for_#qm.qua._dsl.for_.
     """
+
     # Check for array length
     if len(array) == 0:
         raise Exception("The array must be of length > 0.")
     # Check QUA vs python variables
     if not hasattr(var, "_exp"):
         raise Exception("The first argument must be a QUA variable.")
-    if hasattr(array[0], "_exp"):
+    if not isinstance(array[0], (np.generic, int, float)):

```

Comment:
I guess this is ok?
Make sure you test it with all kind of various inputs... Including a np.array (of a np.array!)

Reply:
Yes, I tested it and added the case where it is a bool for good measure...